### PR TITLE
Property table array in subtree format

### PR DESCRIPTION
**Pull request by lilleyse**
Instead of inlining a property table in `tileMetadata` and `contentMetadata` there is now a dedicated section for property tables. This is consistent with [EXT_structural_metadata](https://github.com/CesiumGS/glTF/tree/ext-mesh-features-revision/extensions/2.0/Vendor/EXT_structural_metadata) and is more future proof if we ever need to support property tables referring to other property tables for metadata inheritance (https://github.com/CesiumGS/3d-tiles/issues/641).

### CODE REVIEW

File: extensions/3DTILES_implicit_tiling/schema/subtree/contentMetadata.schema.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
```
Comment (ptrgags) [inline extensions/3DTILES_implicit_tiling/schema/subtree/contentMetadata.schema.json:1]: Do we forsee that `contentMetadata` and tileMetadata` will diverge? the schema files are pretty much identical. Might be okay, just thought I'd ask.

File: extensions/3DTILES_implicit_tiling/README.md [hunk 2]
```diff
@@ -387,7 +387,7 @@ Subtree metadata (`subtreeMetadata`) is encoded in JSON according to the [JSON F
 <!-- omit in toc -->
 #### Property Tables
 
-Binary property values are stored in a **property table**. A property table must specify its class (`class`), which refers to a class ID in the `3DTILES_metadata` extension of root tileset JSON, and a dictionary of properties (`properties`), where each key is a property ID correspond to a class property and each value is the index of the buffer view containing property values. The property table may provide value arrays for only a subset of the properties of its class, but class properties marked `required: true` must not be omitted.
+Binary property values are stored in a **property table**. A property table must specify its class (`class`), which refers to a class ID in the `3DTILES_metadata` extension of root tileset JSON,  a dictionary of properties (`properties`), where each key is a property ID correspond to a class property and each value is the index of the buffer view containing property values, and a count (`count`) for the number of elements in the property table. The property table may provide value arrays for only a subset of the properties of its class, but class properties marked `required: true` must not be omitted.
```
Comment (ptrgags) [inline extensions/3DTILES_implicit_tiling/README.md:390]:
```suggestion
Binary property values are stored in a **property table**. A property table must specify its class (`class`), which refers to a class ID in the `3DTILES_metadata` extension of root tileset JSON, a dictionary of properties (`properties`), where each key is a property ID correspond to a class property and each value is the index of the buffer view containing property values, and a count (`count`) for the number of elements in the property table. The property table may provide value arrays for only a subset of the properties of its class, but class properties marked `required: true` must not be omitted.
```

**Review by ptrgags — COMMENTED** [review]
While I haven't yet started the implementation, this change looks sensible to me.

File: extensions/3DTILES_implicit_tiling/schema/subtree/contentMetadata.schema.json (hunk 1 above)
Comment (lilleyse) [reply extensions/3DTILES_implicit_tiling/schema/subtree/contentMetadata.schema.json:1]: Unlikely that they would diverge, but I think two files is ok for now.

**Review by javagl — COMMENTED** [review]
I think it's good to align the structure more with that of `EXT_structual_metadata`. 
I'm a bit confused about `tileMetadata` and `contentMetadata`, though: They are ... wrappers for an integer. If I had to find a reason for that, could only come up with ~"well, maybe, some day, we want to store other things there, beyond the property table index". But that sounds a bit vague. Is there any _specific_ reason to wrap this index into objects?

**Comment by lilleyse** [comment]
> I'm a bit confused about `tileMetadata` and `contentMetadata`, though: They are ... wrappers for an integer. If I had to find a reason for that, could only come up with ~"well, maybe, some day, we want to store other things there, beyond the property table index". But that sounds a bit vague. Is there any _specific_ reason to wrap this index into objects?

Future proofing was the main reason. Maybe there would be other encodings in the future? Metadata inheritance is also in the back of my mind, but now that I think about it some more those fields would be part of the property table definition. It should be ok to simplify and make them integers instead of wrappers for integers. I'll make the change.

**Comment by lilleyse** [comment]
@javagl updated

**Comment by javagl** [comment]
> Future proofing was the main reason. 

So now it will be my fault if we hit a road block in the future :-) But I don't see any reason now to not merge it.

**Pull request merged**